geopandas/tests/test_geoseries.py

```diff
@@ -434,6 +441,35 @@ def test_data_is_none(self):
         s = GeoSeries(index=range(3))
         check_geoseries(s)
 
+    def test_empty_array(self):
+        # with empty data that have an explicit dtype, we use the fallback or
+        # not depending on the dtype
+        arr = np.array([], dtype="bool")
+
+        # dtypes that can never hold geometry-like data
+        for arr in [
+            np.array([], dtype="bool"),
+            np.array([], dtype="int64"),
+            np.array([], dtype="float32"),
+            # this gets converted to object dtype by pandas
+            # np.array([], dtype="str"),
+        ]:
+            with pytest.warns(FutureWarning):
+                s = GeoSeries(arr)
+            assert not isinstance(s, GeoSeries)
+            assert type(s) == pd.Series
+
+        # dtypes that can potentially hold geometry-like data (object) or
+        # can come from emtpy data (float64)
+        for arr in [
+            np.array([], dtype="object"),
+            np.array([], dtype="float64"),
+            np.array([], dtype="str"),
+        ]:
+            with pytest.warns(None):
+                s = GeoSeries(arr)
```

Comment:
I _think_ that this does nothing. If you want to check that no warning is emitted here, you need

```suggestion
            with pytest.warns(None) as record:
                s = GeoSeries(arr)
            assert not record
```

Reply:
Ah, yes, thanks!